scaling_performance/network_optimization.adoc

```diff
@@ -0,0 +1,142 @@
+[[scaling-performance-routing-network-optimization]]
+= Routing and Network Optimization
+{product-author}
+{product-version}
+:data-uri:
+:icons:
+:experimental:
+
+
+[[scaling-performance-scaling-router-haproxy]]
+== Scaling {product-title} HAProxy Router
+
+The {product-title}
+xref:../install_config/router/index.adoc#install-config-router-overview[router]
+is the ingress point for all external traffic destined for {product-title}
+services.
+
+On an Amazon EC2 m4.xlarge instance, a single HAProxy router is able to handle
+between 7000-32000 HTTP keep-alive requests depending on encryption, page size,
+and the number of connections used. For example, when using TLS
+xref:../architecture/core_concepts/routes.adoc#edge-termination[edge] or
+xref:../architecture/core_concepts/routes.adoc#re-encryption-termination[re-encryption]
+terminations with large page sizes and a high numbers of connections, expect to
+see results in the lower range. With HTTP keep-alive, a single HAProxy router is
+capable of saturating 1 Gbit NIC at page sizes as small as 8 kB.
+
+The table below shows HTTP keep-alive performance on such a public cloud
+instance with a single HAProxy and 100 routes:
+
+[cols="2,3,3"]
+|===
+|Encryption |Page size |HTTP(s) requests per second
+|none |1kB |15435
+|none |4kB |11947
+|edge |1kB |7467
+|edge |4kB |7678
+|passthrough |1kB |25789
+|passthrough |4kB |17876
+|re-encrypt |1kB |7611
+|re-encrypt |4kB |7395
+
+|===
+
+However, on Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50 GHz, you can expect roughly
+twice the performance of the Amazon EC2 m4.xlarge router instance above. This
+overhead is introduced by the virtualization layer in place on public clouds and
+holds mostly true for private cloud-based virtualization as well. The following
+table is a guide on how many applications to use behind the router:
+
+[cols="2,4"]
+|===
+|Number of applications |Application type
+|5-10 |static file/web server or caching proxy
+|100-1000 |applications generating dynamic content
+
+|===
+
+In general, HAProxy can saturate about 10-1000 applications, depending on the
```

Comment:
I initially said 10--1000, but in reality, it can be lower than 10.  If you stick nginx behind HAProxy, a single nginx will easily saturate HAProxy in cases where it serves static content.  In HAProxy documentation, they say 5--10 static file servers/caching proxies, so we should probably be consistent and say 5--1000 applications here as says in the reviewed google doc.